docs/releases.md

```diff
@@ -0,0 +1,149 @@
+- `POINT` can contain bug and security fixes.
+
+No additional markers used to denote pre, alpha, beta or other phases.
+
+## Regular Release Cadence
+
+Cloud Hypervisor is under active development. The regular release schedule is therefore
+quite dynamic and the life time is quite short.
+
+Approximatly every 6 weeks a new major release is issued.
+
+Point releases are issued on demand, when important bugfixes are in the queue. A
+regular release would receive bugfixes for the next two cycles (~12 weeks) and then be
+considered EOL.
+
+```
+++++ Active release support
+E - EOL
+
+        2021                2022                2023 
+         |    |    |    |    |    |    |    |    | 
+18.0     |    |    |  ++++++++E
+19.0     |    |    |    |++++++++E
+20.0     |    |    |    |   ++++++++E
+21.0     |    |    |    |    | ++++++++E
+22.0     |    |    |    |    |    +++++++++E
+23.0     |    |    |    |    |    |  +++++++++E
+
+```
+
+
+## LTS Release Cadence
+
+A new LTS release is promoted from a regular release every 12 months with the
+support timespan of 18 months. This way, there's always a 6 month overlap window 
+suitable for migration.
+
+```
+++++ Active release support
+E - EOL
+
+        2022                2023               2024                 2025                2026
+         |    |    |    |    |    |    |    |    |    |    |    |    |    |    |    |    |
+23.0     |    |+++++++++++++++++++++++++++++E
+43.0     |    |    |    |    |    |+++++++++++++++++++++++++++++E
+63.0     |    |    |    |    |    |    |    |    |    |+++++++++++++++++++++++++++++E
+
+```
+
+## Stablity Considerations
+
+The table below depicts the stability guarantees for regular vs. LTS releases.
+
+Aspect                       | Regular Release           | LTS Release
+-----------------------------|---------------------------|--------------------------------------
+Feature deprecation          | Can be done               | Cannot be done
+Feature add/remove           | Can be done               | Cannot be done
+Dependency upgrade           | Can be done               | Possible, security conditions [^1]
+Rust compatbility            | Can be changed            | Minumum supported edition version must be kept
+KVM compatibility            | Can be changed            | Must be kept
+UEFI compatblity             | Can be changed            | Must be kept
+API compatibility            | Can be changed [^2]       | Must be kept
+Guest OS support             | Can be changed            | Must be kept
+Device model                 | Can be changed            | Must be kept
+Snapshot/restore support     | Not across versions [^3]  | Not across versions [^3]
+Live migration support       | Not across versions [^3]  | Not across versions [^3]
+Experimental features        | Can change substantially [^3] | Can change substantially [^3]
+Source compatibility         | Should be kept, if possible | Must be kept
+
```

Comment:
I think this table is confusing. I see an LTS release as just a normal major release where we are willing to make point releases for longer. At this point in time I don't want to offer any guarantees between LTS releases. 

And your comment about deprecation/removal is confusing since the LTS release is also a regular release that allows deprecations or removals.

I do like the idea of a table but I think that the two columns should be Major Release & Minor release.

Reply:
> I think this table is confusing. I see an LTS release as just a normal major release where we are willing to make point releases for longer.  At this point in time I don't want to offer any guarantees between LTS releases.

I didn't have an intention to talk about differences between two LTS. If anything, probaly there's no guarantee at all, as those will have completely different base and be separated in time quite far. 

> And your comment about deprecation/removal is confusing since the LTS release is also a regular release that allows deprecations or removals.
>

Nope, it is actually the understanding I took from the discussion and also how it is done by other projects. If removals of API are possible in LTS, what is the difference between say a latest major release and LTS?

As a simplified version - LTS would usually mean it is frozen and it is only possible to backport non intrusive bug fixes, necessary small dependency upgrades, etc. This doesn't contradict with "it's a normal major release that receives point releases longer", however it is essential what those point releases contain. Say if in `main` a feature is removed, LTS wolud still contain it and any bugs related to that feature would need to be fixed in LTS. If LTS can receive intrusive changes, it would defeat the meaning of the stability it usually is supposed to bring.

Perhaps the LTS part should be left be till the discussion on the LTS issue is complete.
 
> I do like the idea of a table but I think that the two columns should be Major Release & Minor release.

Perhaps there should be two tables - one explaining LTS vs. major, anothe rexplaining major vs. point. Please feel free also to edit this part directly, it should be possible as far as i see.

Thanks

Reply:
> As a simplified version - LTS would usually mean it is frozen and it is only possible to backport non intrusive bug fixes, necessary small dependency upgrades, etc. This doesn't contradict with "it's a normal major release that receives point releases longer", however it is essential what those point releases contain. Say if in `main` a feature is removed, LTS wolud still contain it and any bugs related to that feature would need to be fixed in LTS. If LTS can receive intrusive changes, it would defeat the meaning of the stability it usually is supposed to bring.

In my view an LTS is just a major release that we will make point releases from for an extended period of time. The same criterion exists for those issues fixed in point releases (critical & security bug fixes). Importantly it would not include any backports of "non intrusive bug fixes" in the general sense.

An LTS would not receive intrusive changes as you say. I think the confusion between us comes as a terminology. I don't consider the point releases from the major declared as the LTS, the LTS releases themselves. They are just point releases so the rules around point releases apply. To me you read the term LTS releases to mean the point releases from the LTS major release.

Hence this is why i'm asserting that the only difference between a non-LTS major release and an LTS one is the length of duration for which point releases may be made. All the usual rules around point releases would still apply.

Reply:
> > As a simplified version - LTS would usually mean it is frozen and it is only possible to backport non intrusive bug fixes, necessary small dependency upgrades, etc. This doesn't contradict with "it's a normal major release that receives point releases longer", however it is essential what those point releases contain. Say if in `main` a feature is removed, LTS wolud still contain it and any bugs related to that feature would need to be fixed in LTS. If LTS can receive intrusive changes, it would defeat the meaning of the stability it usually is supposed to bring.
> 
> In my view an LTS is just a major release that we will make point releases from for an extended period of time. The same criterion exists for those issues fixed in point releases (critical & security bug fixes). Importantly it would not include any backports of "non intrusive bug fixes" in the general sense.
> 

Gotcha. Yes, this is a good scoping then. This means, that backward incompatible changes can still flow in at maintainers discretion, depending on how critical the bug is.  

> An LTS would not receive intrusive changes as you say. I think the confusion between us comes as a terminology. I don't consider the point releases from the major declared as the LTS, the LTS releases themselves. They are just point releases so the rules around point releases apply. To me you read the term LTS releases to mean the point releases from the LTS major release.
> 

I definitely was intended to depict the diff between `LTS` vs. `non-LTS` only. Perhaps here also the discrepancy that my picture of the stability would be a bit stricter, so then it would make an impression it would be about a point release. On the other hand, a possibility to guarantee that level of stability might not exist.

Perhaps talking in some concrete examples, given there would be this version history (omitting point releases for simplicity):

```
41.0 +++++++++++++++++E
+++ ++++42.0 (LTS)++++++++++++++++++++++++++++E
+++++++++++++43.0+++++++++++++++++++E
+++++++++++++++++++44.0 ++++++++++++++
...........................................................................................................
+++++++++++++++++++++++++++++ ++++++++++++55.0 (LTS) +++
++++++++++++++++++++++++++++++++++++ +++++++++++++56.0 .....
```

Removing a CLI option, variant 1:
- Deprecate in 41.0 and add a deprecation warning.
- Release 42.0 LTS - there's a deprecation for the whole lifetime.
- Remove deprecated item in 44.0, no more deprecation warning as well.
- Still the option is present in 42 LTS, at this point:
  - Does this mean the feature is legit to be removed in LTS?

Removing a CLI option, variant 2:
- Release 42.0 LTS - there's no deprecation warning, etc.
- Deprecate in 43.0 and add a deprecation warning.
- 42.0 LTS still has no deprecation for it's lifetime.
- Remove deprecated item in 45.0, no more deprecation warning.
- Still the option is present in 42 LTS.
- Some time after, 55.0 LTS doesn't contain this option which is legit.

What would be a suitable approach to handle the described situation from your POV? AFM the variant 2 would provide the cleanest way, still keeping in mind the maintainer's discretion. Perhaps it's too far fetched, just looking to be as concrete as possible.

Thanks

Reply:
> * Deprecate in 41.0 and add a deprecation warning.
> 
> * Release 42.0 LTS - there's a deprecation for the whole lifetime.
> 
> * Remove deprecated item in 44.0, no more deprecation warning as well.
> 
> * Still the option is present in 42 LTS, at this point:
>   
>   * Does this mean the feature is legit to be removed in LTS?

No. Because again, an LTS is just a major release with point releases for longer. We would never remove functionality in a point release. Again I think it's unhelpful to think of those point releases as the LTS releases instead think of the major release as the LTS release. Whatever functionality is in will stay there for for the whole lifetime.

I would summarise my view as:

"An LTS release is just a major release for which point releases are made for longer following the same rules for what can be backported to a point release."

Reply:
Thanks for the clarifications! I now reworked this part by removing the table and adding "security considerations" subparagraph to both regular and lts sections. Also removed the mentions of source compat and some others that we've clarified to be not relevant. also, moved the Rust compat to the basic terms list. Regarding the point release, it's otherwise already told at some other place that it's "bug and security", so only mentioning "critical bug and security" in the LTS part. Perhaps there's still more to tell, but this at least gives an impression to the consumer, what to expect. Some corrections would probably be still needed, once the LTS approach has been decided.

Thanks